### ISSUE TITLE
NVDA Does Not Read Multiple Lines in a Read Only Textarea HTML Element in Chrome 98

### ISSUE THREAD

**Issue by frastlin** [nvaccess/nvda#13369]
### Steps to reproduce:

1. Using Chrome 98+ and NVDA 2021.3.1 visit:
https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_textarea_readonly
2. Hit the E key until you reach the read only example and hit enter to enter focus mode
3. Down arrow to hear the multiple lines
4. Notice that the second line is not read.

### Actual behavior:
 NVDA does not read the second line in the read only Textarea Element in CHrome.

### Expected behavior:

NVDA should read multiple lines as it does in Firefox.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:

2021.3.1

#### Windows version:

10 64 bit

#### Name and version of other software in use when reproducing the issue:

Chrome Version 98.0.4758.102 (Official Build) (64-bit)

#### Does the issue still occur after restarting your computer?

Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

No.

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes.

**Comment by Adriani90** [comment]
cc: @ObjectInSpace, @Aleventhal

**Comment by MarcoZehe** [comment]
I can confirm this. Furthermore, you can't even rightArrow onto the next character, the cursor doesn't move from the "A" of the word "At".

**Comment by frastlin** [comment]
If someone could test to see if Jaws has the same problem, we can figure out if this is an NVDA or Chrome bug. Chrome may have disabled the cursor from moving in read-only fields.

**Comment by Yegorich555** [comment]
Issue exists also for contenteditable
```HTML
<label for="wup5">
  <span>
    <span contenteditable="true" role="textbox" aria-labelledby="wup5" style="white-space: pre-wrap">
      there is multiline text
      placed in contenteditable
    </span>
    <strong id="wup5">Test NVDA</strong>
  </span>
</label>
```

**Comment by Adriani90** [comment]
I can reproduce this particular issue with NVDA 2024.4.1 in Chromium canary 133 but not in Firefox. However, I cannot reproduce the issue in the contentEditable as stated in https://github.com/nvaccess/nvda/issues/13369#issuecomment-1447695512

It seems however, that in contentEditable multiine text doesn't add linefeed characters, so it seems NVDA and also Jaws merge the last word from one line with the first word from the next line. This makes reading in contentEditables quite challenging. What is the reason for that? 

cc: @jcsteh for Firefox, is there a bug already filed regarding this?

I also created two Chromium bugs:
1. For textarea readonly atribute
https://issues.chromium.org/issues/382291453

2. For contentEditable (it needs probably also a Firefox bug report)
https://issues.chromium.org/issues/382294241

**Comment by jcsteh** [comment]
> It seems however, that in contentEditable multiine text doesn't add linefeed characters, so it seems NVDA and also Jaws merge the last word from one line with the first word from the next line.

I don't understand. Steps to reproduce/expected results/actual results please?

**Comment by Adriani90** [comment]
Str for Firefox:
1. With NVDA or Jaws running, open following test case:
data:text/html,<label for="wup5"><span contenteditable="true" role="textbox" aria-labelledby="wup5" style="white-space: pre-wrap">there is multiline textplaced in contenteditable</span><strong id="wup5">Test NVDA</strong></label>
2. Press enter on the edit field to switch to focus mode
3. Navigate the text word by word with ctrl+arrow left and right, or by character with arrow left and right

Actual:
The words "text" and "placed" are merged. This makes reading texts in contentEditables with screen readers very challenging.

Expected behavior:
Linefeed character should be added to contentEditables when multiline text is added (e.g. via white-space: pre-wrap). The system caret should be able to navigate the linefeed characters, so that words from different lines are not merged.

It seems in Chrome is rather the problem that the white-space: pre-wrap is not considered either, it renders only a space character between the words instead of a linefeed character.

**Comment by jcsteh** [comment]
Your test case doesn't wrap across two lines here. I assume the intention is that it should?

**Comment by Adriani90** [comment]
Ah I managed to adjust the test case to cause a wrap with white-space: pre-wrap. This seems to work in Firefox. I just didn't have enough text in the box:

```
data:text/html,<label for="wup5"><span contenteditable="true" role="textbox" aria-labelledby="wup5" style="white-space: pre-wrap">         there is enough multiline text       in this world           to produce a line long enough in order to be placed and wrapped in multi line contenteditable. Or maybe not? Who knows? Let's find out.     Hello world Hello world Hello world Hello world Hello world Hello world Hello world Hello world Hello world Hello world Hello world </span><strong id="wup5">Test NVDA</strong> </label>
```

It seems NVDA indeed reports blank when arrowing to the next line by character, in Chrome it reports "space" instead of blank.

The chromium bug remains valid.

**Comment by Adriani90** [comment]
Regarding readonly text areas, NVDA 2025.3.1 and Chromium 142, navigation in focus mode still does not work. Here is the last comment from chromium devs:

> This is something that can be implemented on the screen reader side by turning off forms mode and going into virtual buffer mode. It's a relatively rare case.
> If there's something in the standard that says that the browser needs to implement caret nav in a readonly field, please link that.

In firefox this works as expected though.
@jcsteh, @SaschaCowley are there any WCAG specs for caret navigation in readonly text areas? I could not find any, but i also cannot imagine that noone thought about this before.
Especially in web applications where text areas can be protected or unprotected (e.g. Microsoft Word Online web app), readonly text areas get very relevant and as an user you don't want to always switch back and forth between focus mode and browse mode.
When a read only text area contains alot of text which does not fit completely on the screen, it is even not possible to read the complete content because in browse mode the screen doesn't scroll. So only a part of the text is navigable.
